src/devices/Lp55231/Lp55231.cs

```diff
@@ -0,0 +1,174 @@
+    public class Lp55231 : IDisposable
+    {
+        /// <summary>
+        /// Default I2C address of Motor Hat
+        /// </summary>
+        public const int DefaultI2cAddress = 0x32;
+
+        /// <summary>
+        /// Gets the channel of the red element of the specific instance
+        /// </summary>
+        /// <param name="instance">The RGB instance (0-2)</param>
+        /// <returns>The channel number</returns>
+        internal static byte RedChannel(byte instance) => (byte)(6 + instance);
+
+        /// <summary>
+        /// Gets the channel of the green element of the specific instance
+        /// </summary>
+        /// <param name="instance">The RGB instance (0-2)</param>
+        /// <returns>The channel number</returns>
+        internal static byte GreenChannel(byte instance) => (byte)(instance * 2);
+
+        /// <summary>
+        /// Gets the channel of the blue element of the specific instance
+        /// </summary>
+        /// <param name="instance">The RGB instance (0-2)</param>
+        /// <returns>The channel number</returns>
+        internal static byte BlueChannel(byte instance) => (byte)(instance * 2 + 1);
+
+        private readonly I2cDevice _device;
+        private readonly IReadOnlyList<RgbLed> _rgbLeds;
+        private readonly bool _disposeDevice;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Lp55231"/> class with the specified I2C settings.
+        /// </summary>
+        /// <param name="device">The I2C device to communicate with the Lp55231.</param>
+        /// <remarks>
+        /// The default i2c address is 0x60, but the HAT can be configured in hardware to any address from 0x60 to 0x7f.
+        /// The PWM hardware used by this HAT is a PCA9685. It has a total possible frequency range of 24 to 1526 Hz.
+        /// Setting the frequency above or below this range will cause PWM hardware to be set at its maximum or minimum setting.
+        /// </remarks>
+        public Lp55231(I2cDevice device)
+        {
+            _device = device;
+
+            _rgbLeds = new[]
+            {
+                new RgbLed(0, SetIntensity),
+                new RgbLed(1, SetIntensity),
+                new RgbLed(2, SetIntensity)
+            };
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Lp55231"/> class with the specified I2C settings.
+        /// </summary>
+        /// <param name="settings">The I2C settings of the Lp55231.</param>
+        /// <remarks>
+        /// The default i2c address is 0x60, but the HAT can be configured in hardware to any address from 0x60 to 0x7f.
+        /// The PWM hardware used by this HAT is a PCA9685. It has a total possible frequency range of 24 to 1526 Hz.
+        /// Setting the frequency above or below this range will cause PWM hardware to be set at its maximum or minimum setting.
+        /// </remarks>
+        public Lp55231(I2cConnectionSettings? settings = default)
```

Comment:
you should not have this constructor. Pattern is not to create an I2C device in constructors. The bus ID can vary, the I2cDevice can be attached to another type of hardware. There are so many cases so that's the reason we are not doing it.

Reply:
Removed.